tests/phpunit/tests/dependencies/styles.php

```diff
@@ -628,4 +628,54 @@ public function test_wp_maybe_inline_styles_no_path() {
 
 		$this->assertSame( $GLOBALS['wp_styles']->registered['test-handle']->src, $url );
 	}
+
+	/**
+	 * Tests detecting nested tags in inline styles.
+	 *
+	 * @ticket 61828
+	 * @dataProvider data_styles_with_nested_tags
+	 */
+	public function test_wp_add_inline_style_with_nested_style_tags( $expected, $styles, $correct_usage = false ) {
+		if ( ! $correct_usage ) {
+			$this->setExpectedIncorrectUsage( 'wp_add_inline_style' );
+		}
+		wp_enqueue_style( 'handle', 'http://example.com', array(), 1 );
+		wp_add_inline_style( 'handle', $styles );
+
+		$this->assertSame( $expected, get_echo( 'wp_print_styles' ), "Inline styles - $styles - do not match expected values." );
+	}
+
+	public function data_styles_with_nested_tags() {
+		return array(
+			array(
+				"<link rel='stylesheet' id='handle-css' href='http://example.com?ver=1' type='text/css' media='all' />\n<style id='handle-inline-css' type='text/css'>\n\\3c\\2fstyle><script>alert(0)</script><style>\n</style>\n",
+				'</style><script>alert(0)</script><style>',
+			),
+			array(
```

Comment:
```suggestion
			'closed_and_reopened' => array(
```

And names for each of the others too. Makes it easier to figure out what happens during failures.